src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -5712,6 +5768,9 @@ QGISEXTERN QString loadStyle( const QString &uri, QString &errCause )
   QgsDataSourceUri dsUri( uri );
   QString sqlitePath = dsUri.database();
   QgsDebugMsg( "Database is: " + sqlitePath );
+  // Avoid sql-statement errors
+  if ( dsUri.schema().isEmpty() )
+    return false;
```

Comment:
Where is this being used?
When dsUri.schema() is empty, it caused sql-errors.
The same for saveStyle.